File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int M = Integer.parseInt(br.readLine().trim());
        HashSet<Integer> S = new HashSet<>();
        String[] line;
        for (int i = 0; i < M; i++) {
            line = br.readLine().split(" ");
            if (line[0].equals("add")) {
                S.add(Integer.parseInt(line[1]));
            } else if (line[0].equals("remove")) {
                S.remove(Integer.parseInt(line[1]));
            } else if (line[0].equals("check")) {
                if (S.contains(Integer.parseInt(line[1]))) {
                    sb.append(1).append('\n');
                } else {
                    sb.append(0).append('\n');
                }
            } else if (line[0].equals("toggle")) {
                int num = Integer.parseInt(line[1]);
                if (S.contains(num)) {
                    S.remove(num);
                } else {
                    S.add(num);
                }
            } else if (line[0].equals("empty")) {
                S.clear();
            } else if (line[0].equals("all")) {
                S.clear();
                for (int j = 1; j <= 20; j++) {
                    S.add(j);
                }
            }
        }
        System.out.print(sb.toString());
        br.close();
    }
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

M = int(input())
S = dict()
for i in range(21):
    S[i] = 0

for _ in range(M):
    yeonSan = input().split()
    if len(yeonSan) == 2:
        func, num = yeonSan[0], int(yeonSan[1])
        if func == "add":
            S[num] = 1
        elif func == "remove":
            S[num] = 0
        elif func == "check":
            if S[num] == 0:
                print(0)
            elif S[num] == 1:
                print(1)
        elif func == "toggle":
            if S[num] == 0:
                S[num] = 1
            elif S[num] == 1:
                S[num] = 0
    elif yeonSan[0] == "empty":
        for i in range(21):
            S[i] = 0
    elif yeonSan[0] == "all":
        for i in range(21):
            S[i] = 1